code/modules/atmospherics/components/binary_devices/circulator.dm

```diff
@@ -90,63 +93,35 @@
 
 	return 1
 
-/obj/machinery/atmospherics/binary/circulator/attackby(obj/item/W, mob/user)
-	if(isWrench(W))
-		playsound(src.loc, 'sound/items/Ratchet.ogg', 75, 1)
-		anchored = !anchored
-		user.visible_message("[user.name] [anchored ? "secures" : "unsecures"] the bolts holding [src.name] to the floor.", \
-					"You [anchored ? "secure" : "unsecure"] the bolts holding [src] to the floor.", \
-					"You hear a ratchet")
-
-		if(anchored)
-			temperature_overlay = null
-			if(dir & (NORTH|SOUTH))
-				initialize_directions = NORTH|SOUTH
-			else if(dir & (EAST|WEST))
-				initialize_directions = EAST|WEST
-
-			atmos_init()
-			build_network()
-			if (node1)
-				node1.atmos_init()
-				node1.build_network()
-			if (node2)
-				node2.atmos_init()
-				node2.build_network()
-		else
-			if(node1)
-				node1.disconnect(src)
-				qdel(network1)
-			if(node2)
-				node2.disconnect(src)
-				qdel(network2)
-
-			node1 = null
-			node2 = null
-		update_icon()
-
+/obj/machinery/atmospherics/binary/circulator/wrench_floor_bolts(mob/user, delay)
+	. = ..()
+	if(anchored)
+		temperature_overlay = null
+		if(dir & (NORTH|SOUTH))
+			initialize_directions = NORTH|SOUTH
```

Comment:
this bloc seems to be repeating in every binary machine, can't it be moved to `/obj/machinery/atmospherics/binary/wrench_floor_bolts` ?